plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleConnectorTest.java

```diff
@@ -411,6 +412,60 @@ public void testTooLargeDomainCompactionThreshold()
                 "SELECT * from nation", "Domain compaction threshold \\(10000\\) cannot exceed 1000");
     }
 
+    @Override
+    public void testNativeQuerySimple()
+    {
+        // override because Oracle requires the FROM clause, and it needs explicit type
+        assertThat(query("SELECT * FROM TABLE(system.query(query => 'SELECT CAST(1 AS number(2, 1)) FROM DUAL'))"))
+                .matches("VALUES 1.0");
+    }
+
+    @Override
+    public void testNativeQuerySelectFromNation()
+    {
+        // override because Oracle does not use tpch schema
+        assertThat(query("SELECT * FROM TABLE(system.query(query => 'SELECT name FROM nation WHERE nationkey = 0'))"))
+                .matches("VALUES CAST('ALGERIA' AS varchar(25))");
+    }
+
+    @Override
+    public void testNativeQuerySelectFromTestTable()
+    {
+        // override because Oracle does not support type bigint
+        try (TestTable testTable = simpleTable()) {
+            assertThat(query(format("SELECT * FROM TABLE(system.query(query => 'SELECT * FROM %s'))", testTable.getName())))
+                    .matches("VALUES 1.0, 2.0");
+        }
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableDoesNotExist()
+    {
+        // override because Oracle succeeds in preparing query, and then fails because of no metadata available
+        assertFalse(getQueryRunner().tableExists(getSession(), "non_existent_table"));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'INSERT INTO non_existent_table VALUES (1)'))"))
+                .hasMessageContaining("Statement not supported: INSERT INTO non_existent_table VALUES (1). ResultSetMetaData not available");
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableExists()
+    {
+        // override because Oracle does not support type bigint
+        try (TestTable testTable = simpleTable()) {
+            assertThatThrownBy(() -> query(format("SELECT * FROM TABLE(system.query(query => 'INSERT INTO %s VALUES (3)'))", testTable.getName())))
+                    .hasMessageContaining(format("Statement not supported: INSERT INTO %s VALUES (3). ResultSetMetaData not available", testTable.getName()));
+            assertThat(query("SELECT * FROM " + testTable.getName()))
+                    .matches("VALUES 1.0, 2.0");
+        }
+    }
+
+    @Override
+    protected TestTable simpleTable()
+    {
+        // override because Oracle does not support type bigint, and it does not use tpch schema
```

Comment:
> and it does not use tpch schema

in the method being overriden, you can replace `tpch` with `getSession().getSchema().orElseThrow()`